### ISSUE TITLE
Is there any version of OneTimeSetUpAttribute with parameters ?

### ISSUE THREAD

**Issue by rajmondburgaj** [nunit/nunit#3004]
So, first I am describing my issue in order to explain why would I need such a feature.

I set my platformType into a method marked with [OneTimeSetp] attribute like below:

        [OneTimeSetUp]
        public void Configure() 
        {
            TestPlatform.SetPlatform(PlatformType.MainLine);
            PosApiMapper.Initialize();
            Mapper.AssertConfigurationIsValid();
        }

Depending on platformType then I execute all unit tests in the project. But since I have many platform types I need manually to change this and then re-run the whole project again. I know I can set the platform type in each testing method as a param but that would make the code ugly to my opinion and furthermore the real problem relies on this part `PosApiMapper.Initialize();` which makes the mapping logic using `Automapper`. So, the mappings are different for each platform type. I want to ask if there is a version of `[OneTimeSetUp]` attribute which accepts params already ? If not is there any way to use inheritance or something to make it work as expected ?

So, the idea is to run `Configure` method with a param, second step is to run the whole project, third step would be to run the method marked with [OneTimeTearDown] attribute and then we start again from beginning, so, run `Configure` method with another param, re-run project again and so on..  

[Here](https://stackoverflow.com/questions/52185708/is-there-any-ontimesetup-attribute-which-accepts-parameters-in-nunit) is my link to SO.

**Comment by mikkelbu** [comment]
Hi @rajmondburgaj 
How are you executing the tests? Because my initial idea was to pass the information about PlatformType from the nunit-console - via a parameter - to the test project and then execute the nunit-console with different parameters.

**Comment by rajmondburgaj** [comment]
Currently I use the interface from Visual Studio and RSharper but I can try this out, anyway it does not completely solve the issue as you have to always run all tests using nunit-console which to be honest is not the best solution to my opinion.

**Comment by CharliePoole** [comment]
We discussed this on SO and I tried to explain to @rajmondburgaj that the specific feature wasn't a great fit with existing NUnit features. I suggested he ask for parameterized SetUpFixtures instead.

Why not parameterized OneTimeSetUp methods?

1. The attribute is not a fixture builder, but only a marker.

2. Multiple instances are all called in an indeterminate order for the same fixture.

3. The attribute is used in TestFixtures as well, where any replication would multiply the impact of parameters.

4. Parameterized fixtures is how NUnit already accomplishes the same thing, just not for SetUpFixtures.
